spotify.go

```diff
@@ -124,6 +171,78 @@ func decodeError(r io.Reader) error {
 // authenticate, you can use `DefaultClient`.
 type Client struct {
 	http *http.Client
+
+	autoRetry     bool
+	retryDuration time.Duration
+}
+
+func (c *Client) SetAutoRetry(flag bool) {
+	c.autoRetry = flag
+}
+
+// executeOpt executes a non-GET request. `needsStatus` describes another code
+// that can represent success. Note that in all current usages of this function,
+// we need to still allow a 200 even if we'd also like to check for a second
+// success code.
+func (c *Client) executeOpt(req *http.Request, needsStatus int, result interface{}) error {
```

Comment:
Although technically this might be useful, there are only a handful of calls to `executeOpt` (mostly in playlist.go) and none of them check for more than one other success-code. Plus, we have many implementations of an "opt" function allowing for extended usage and a "non-opt" function allowing for a simpler, common usage. I believe that this function makes the most sense as is, but, if you insist, I'll make the change.